ext/src/http/client/curl/http_operation_curl.cc

```diff
@@ -410,9 +476,145 @@ CURLcode HttpOperation::Setup()
+      return CURLE_UNKNOWN_OPTION;
+#endif
+    }
+
+    /* 2 - CLIENT KEY */
+
+    if (!ssl_options_.ssl_client_key_path.empty())
+    {
+      const char *path = ssl_options_.ssl_client_key_path.c_str();
+      curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_SSLKEY, path);
+      curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_SSLKEYTYPE, "PEM");
+    }
+    else if (!ssl_options_.ssl_client_key_string.empty())
+    {
+#if LIBCURL_VERSION_NUM >= 0x074700
+      const char *data = ssl_options_.ssl_client_key_string.c_str();
+      size_t data_len  = ssl_options_.ssl_client_key_string.length();
+
+      struct curl_blob stblob;
+      stblob.data  = const_cast<char *>(data);
+      stblob.len   = data_len;
+      stblob.flags = CURL_BLOB_COPY;
+      curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_SSLKEY_BLOB, &stblob);
+      curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_SSLKEYTYPE, "PEM");
+#else
+      // CURL 7.71.0 (0x07 0x47 0x00) required for CURLOPT_SSLKEY_BLOB.
+      return CURLE_UNKNOWN_OPTION;
+#endif
+    }
+
+    /* 3 - CLIENT CERT */
+
+    if (!ssl_options_.ssl_client_cert_path.empty())
+    {
+      const char *path = ssl_options_.ssl_client_cert_path.c_str();
+      curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_SSLCERT, path);
+      curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_SSLCERTTYPE, "PEM");
+    }
+    else if (!ssl_options_.ssl_client_cert_string.empty())
+    {
+#if LIBCURL_VERSION_NUM >= 0x074700
+      const char *data = ssl_options_.ssl_client_cert_string.c_str();
+      size_t data_len  = ssl_options_.ssl_client_cert_string.length();
+
+      struct curl_blob stblob;
+      stblob.data  = const_cast<char *>(data);
+      stblob.len   = data_len;
+      stblob.flags = CURL_BLOB_COPY;
+      curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_SSLCERT_BLOB, &stblob);
+      curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_SSLCERTTYPE, "PEM");
+#else
+      // CURL 7.71.0 (0x07 0x47 0x00) required for CURLOPT_SSLCERT_BLOB.
+      return CURLE_UNKNOWN_OPTION;
+#endif
+    }
+
+    /* 4 - TLS */
+
+    long min_ssl_version = 0;
+
+    if (!ssl_options_.ssl_min_tls.empty())
+    {
+      min_ssl_version = parse_min_ssl_version(ssl_options_.ssl_min_tls);
+
+      if (min_ssl_version == 0)
+      {
+        return CURLE_UNKNOWN_OPTION;
+      }
+    }
+
+    long max_ssl_version = 0;
+
+    if (!ssl_options_.ssl_max_tls.empty())
+    {
+      max_ssl_version = parse_max_ssl_version(ssl_options_.ssl_max_tls);
+
+      if (max_ssl_version == 0)
+      {
+        return CURLE_UNKNOWN_OPTION;
+      }
+    }
+
+    long version_range = min_ssl_version | max_ssl_version;
+    if (version_range != 0)
+    {
+      curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_SSLVERSION, version_range);
+    }
+
+    /* 5 - CIPHER_LIST */
+
+    if (!ssl_options_.ssl_cipher_list.empty())
+    {
+      // TODO: Unclear when to use CURLOPT_SSL_CIPHER_LIST versus CURLOPT_TLS13_CIPHERS
```

Comment:
Looks that OpenSSL uses separate cipher list for TLS1.3, so people who wants both 1.2 and 1.3 have to provide both ciphers list. This would require separate options in OTLP libs too.

Reply:
@sirzooro 

Exposing a separate 1.2 and 1.3 cipher list option in the configuration is not likely to be accepted in the opentelemetry-specification.

I am experimenting with a possible mitigation, to use both when both 1.2 and 1.3 are needed.
If it works, all the better. If not, this may be a remaining limitation to resolve later.

Cases for only 1.2 or only 1.3 are trivial, and will use only the same cipher_list variable.

Reply:
> Looks that OpenSSL uses separate cipher list for TLS1.3, so people who wants both 1.2 and 1.3 have to provide both ciphers list. This would require separate options in OTLP libs too.

Resolved, with separate options for TLS <= 1.2 and TLS 1.3.